cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -288,6 +291,151 @@ class StdStringAt extends TaintFunction {
   }
 }
 
+/**
+ * The `std::basic_istream` template class.
+ */
+class StdBasicIStream extends TemplateClass {
+  StdBasicIStream() { this.hasQualifiedName("std", "basic_istream") }
+}
+
+/**
+ * The `std::istream` function `operator>>` (defined as a member function).
+ */
+class StdIStreamIn extends DataFlowFunction, TaintFunction {
+  StdIStreamIn() { this.hasQualifiedName("std", "basic_istream", "operator>>") }
+
+  override predicate hasDataFlow(FunctionInput input, FunctionOutput output) {
+    // flow from qualifier to return value
+    input.isQualifierObject() and
```

Comment:
Why is this the qualifier _object_ and not the _address_? I think all the overloads in https://en.cppreference.com/w/cpp/io/basic_istream/operator_gtgt return references.

Reply:
We're returning `*this` (the qualifier object) as a reference (which makes it an address), so I think you're right it should be `isQualifierAddress`.  It looks like there are a bunch of places to fix...

Reply:
... in fact there were only two instances of the problem.  Fixed and made the comments a bit more useful if we want to find these cases in future.